matlab-src/mex-wholebodymodel_balancing/forwardDynamics.m

```diff
@@ -43,41 +40,32 @@
 %% fixing the world reference frame
 qT         = [x_b; qt_b];
 [~,R_b]    = frame2posrot(qT);
- 
-wbm_setWorldFrame(R_b,x_b,[0 0 -9.81]');
 
-wbm_updateState(qj,dqj,[dx_b;omega_W]);
-
-%% MexWholeBodyModel calls
-% warning: there's an issue with the optimized mode: for now, it won't be
+%% mexWholeBodyModel calls
+% warning: there's an issue with the wbm optimized mode: for now, it won't be
 % used
-% M = wbm_massMatrix(); 
-% h = wbm_generalisedBiasForces();
-% H = wbm_centroidalMomentum();
-
-R_binv = eye(3)/R_b;
+R_b_tr = R_b';
 
-M      = wbm_massMatrix(R_binv,x_b,qj); 
-h      = wbm_generalisedBiasForces(R_binv,x_b,qj,dqj,[dx_b;omega_W]);
-H      = wbm_centroidalMomentum(R_binv,x_b,qj,dqj,[dx_b;omega_W]);
+M      = wbm_massMatrix(R_b_tr,x_b,qj); 
```

Comment:
From https://github.com/robotology/mex-wholebodymodel/pull/57/files#diff-b5bd320821a3ac71838bcc9778a3f583R163 I guess that `R_b_tr` is the rotation transforming a vector expressed in the world frame to one expressed in the base frame: 
~~~
v_b = b_R_w * v_w , b_R_w = R_b_tr
~~~

The problem is that wbm function are expected to get in input the `w_R_b` rotation instead (see http://wiki.icub.org/codyco/dox/html/dynamics_notation.html).
Apparently this is working because there is a bug in almost all the code handling Rotations (https://github.com/robotology/mex-wholebodymodel/search?utf8=%E2%9C%93&q=wbi%3A%3ARotation+tempRot%28tempR%29%3B). The rotation matrix is not properly converted from column-major to row-major when passing it from Matlab to WBI . This bug is equivalent to a transposition of the matrix, and this is the reason why passing the transpose of what should be put in input is working. 

Apparently the bug was solved for the setWorldFrame function : https://github.com/robotology/mex-wholebodymodel/blob/efc77a2feb1b8ec13f9693770cf3c1ce07cc2f1a/mex-wholebodymodel/library/src/modelsetworldframe.cpp#L88 , so we need to propagate the fix to all other functions, I will open an issue for that.